src/controllers/cartController.ts

```diff
@@ -1,144 +1,206 @@
 import { Request, Response } from "express"
 import { CartsModel } from "../models/cart";
 import { ProductsModel } from "../models/products";
+import { cartsSchema } from "../validators/validations"
+
 
 export const getCartByUserId = async (req: Request, res: Response) => {
-    try {
-      const userId = parseInt(req.params.id);
-  
-      if (isNaN(userId)) {
-        return res.status(400).json({ error: 'Invalid user ID' });
-      }
-  
-      const carts = await CartsModel.findAll({ where: { user_id: userId } });
-  
-      if (!carts || carts.length === 0) {
-        return res.status(404).json({ error: 'No cart entries found for the specified user ID' });
```

Comment:
ERROR 404 indicates page not found, when there is item that is not found return status 204

Reply:
The HTTP status code 404 is commonly used to indicate that the requested resource could not be found on the server. It is suitable for situations where a particular item or resource is expected to exist, but the server cannot find it.